sdks/java/core/src/main/java/org/apache/beam/sdk/util/ReduceFnRunner.java

```diff
@@ -580,14 +581,16 @@ public void onTimer(TimerData timer) throws Exception {
           "ReduceFnRunner.onTimer: Note that timer {} is for non-ACTIVE window {}", timer, window);
     }
 
-    // If this is an end-of-window timer then, we need to set a GC timer
+    // If this is an end-of-window timer then we'll need to set a garbage collection timer.
```

Comment:
Either:
`we'll need to set a garbage collection timer if allowed lateness is non-zero`
or
`If this is an end-of-window timer and we make it here, then allowed lateness is non-zero, so we 'll need ...`